thirdeye/thirdeye-frontend/app/utils/utils.js

```diff
@@ -275,11 +277,68 @@ export function buildBounds(series, baseline, timeseries, useCurrent) {
  * @param {Array} timeSeries - time series to modify
  */
 export function stripNonFiniteValues(timeSeries) {
-  return timeSeries.map(value => {
-    return (isFinite(value) ? value : null);
+  return timeSeries.map((value) => {
+    return isFinite(value) ? value : null;
   });
 }
 
+/*
+ * Detect if the input is in Object form
+ *
+ * @param {Any} input
+ *   The input to test.
+ *
+ * @returns {Boolean}
+ *   True if input is an object, false otherwise.
+ */
+export function isObject(input) {
+  return input instanceof Object && input.constructor === Object;
+}
+
+/*
+ * Search the input for the existence of the search term
+ *   -Supports searching on following input types - string, integer, float, boolean, object, array
+ *
+ * @param {Any} input
+ *   The input to test.
+ * @param {String} filterStr
+ *   The search term
+ *
+ * @returns {Boolean}
+ *   True if the match is found, false otherwise.
+ */
+export function checkForMatch(input, filterStr) {
+  if (filterStr === '') {
+    return true;
+  }
+
+  if (typeof input === 'string') {
+    return input.includes(filterStr);
+  } else if (typeof input === 'number' || typeof input === 'boolean') {
+    return input.toString().includes(filterStr);
+  } else if (isObject(input)) {
+    for (const prop in input) {
+      if ({}.propertyIsEnumerable.call(input, prop)) {
+        if (checkForMatch(prop, filterStr) || checkForMatch(input[prop], filterStr)) {
```

Comment:
multiple level nest conditions will muddy up code's control flow. Also, it takes difficulty for later maintenance and result devs to spend more time on understanding state and logic. These two `if` statement are unnesstary, I suggest u can put them in one such as `if ({}.propertyIsEnumerable.call(input, prop) && (checkForMatch(prop, filterStr) || checkForMatch(input[prop], filterStr))`

Reply:
I can club the `if's` but I don't necessarily agree it leads to improved maintenance or less time understanding the logic, as the logic is not really changed by clubbing the `if's`.